modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessorTestCase.java

```diff
@@ -138,6 +151,12 @@ public ProcessingType getInnerProcessingType() {
     };
   }
 
+  @Override
+  protected CoreEvent.Builder getEventBuilder() {
```

Comment:
instead of overriding here, could you make this change to apply to all calls to `MuleTestUtils#getTestFlow()`? That would benefit the whole tests execution i think.

Reply:
I guess we could. But I would prefer to make that into a separate PR. Because it may have hidden side effects that we are not considering. For example, I'm seeing that at some point we [added something to keep track of test flows](https://github.com/mulesoft/mule/blob/master/tests/unit/src/main/java/org/mule/tck/junit4/AbstractMuleTestCase.java#L308), which implies that it is not expected for `MuleTestUtils#getTestFlow` to return always the same cached instance.

On the other hand, we do have things like [this one](https://github.com/mulesoft/mule/blob/master/tests/unit/src/main/java/org/mule/tck/util/MuleContextUtils.java#L369), where we create a new test flow and we register it in the registry. But since the test flow always has the same name, calling that method multiple times will override the previous ones in the registry. Only the last one will undergo the lifecycle transitions of the registry.

Reply:
mm, that would mean that only the last one to be registered will be stopped/disposed?

Reply:
Seems like it, though I haven't actually verified it. However, we added [this](https://github.com/mulesoft/mule/commit/1109cc25db6a7bebe3534bdbcd582f73a0aa389c) some time ago, I'm guessing it is related. And it will probably still be needed regardless, for tests that do not use any MuleContext.